docs/running_lbann.rst

```diff
@@ -1,96 +1,434 @@
+
+   + A timestamped work directory will be created each time LBANN is
+     run. The default location of these work directories can be set
+     with the environment variable :bash:`LBANN_EXPERIMENT_DIR`.
+
+   + Supported job managers are Slurm and LSF.
+
+   + LLNL users may prefer to use
+   :python:`lbann.contrib.lc.launcher.run`. This is a wrapper around
+   :python:`lbann.run`, with defaults and optimizations specifically
+   for LC systems.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A simple example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    import lbann
+
+    # ----------------------------------
+    # Construct layer graph
+    # ----------------------------------
 
-------------------------------
-Running the inference engine
-------------------------------
+    # Input data
+    input = lbann.Input()
+    image = lbann.Identity(input)
+    label = lbann.Identity(input)
 
-This section is under construction, requiring input from other team
-members. Until it is complete, please ask questions on the
-`issue tracker <https://github.com/llnl/lbann/issues>`_.
+    # Softmax classifier
+    y = lbann.FullyConnected(image, num_neurons = 10, has_bias = True)
+    pred = lbann.Softmax(y)
+
+    # Loss function and accuracy
+    loss = lbann.CrossEntropy([pred, label])
+    acc = lbann.CrossEntropy([pred, label])
+
+    # ----------------------------------
+    # Setup experiment
+    # ----------------------------------
+
+    # Setup model
+    mini_batch_size = 64
+    num_epochs = 5
+    model = lbann.Model(mini_batch_size,
+                        num_epochs,
+                        layers=lbann.traverse_layer_graph(input),
+                        objective_function=loss,
+                        metrics=[lbann.Metric(acc, name='accuracy', unit='%')],
+                        callbacks=[lbann.CallbackPrint(), lbann.CallbackTimer()])
+
+    # Setup optimizer
+    opt = lbann.SGD(learn_rate=0.01, momentum=0.9)
+
+    # Load data reader from prototext
+    import google.protobuf.text_format as txtf
+    data_reader_proto = lbann.lbann_pb2.LbannPB()
+    with open('path/to/lbann/model_zoo/data_readers/data_reader.prototext', 'r') as f:
+        txtf.Merge(f.read(), data_reader_proto)
+    data_reader_proto = data_reader_proto.data_reader
+
+    # ----------------------------------
+    # Run experiment
+    # ----------------------------------
+
+    lbann.run(model, data_reader_proto, opt)
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Useful submodules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+^^^^^^^^^^^^^^^^^^^^^^^^
+:python:`lbann.modules`
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+A :python:`Module` is a pattern of layers that can be applied multiple
+times in a neural network. Once created, a `Module` is *callable*,
+taking a layer as input and returning a layer as output. They will
+create and manage `Weights` es internally, so they are convenient for
+weight sharing between different layers. They are also useful for
```

Comment:
Is the internal management of weights the advantage over just creating a Python function to call, for example, `lbann.Relu` followed by `lbann.Convolution`?

Reply:
Yep. I was also thinking of recurrent networks, where the layers created when calling a module need to be connected to layers from previous calls (although I'm tempted to reimplement stuff to avoid this kind of behavior).